contracts/hydro/src/token_manager.rs

```diff
@@ -318,22 +334,62 @@ impl TokenInfoProviderLSM {
     }
 }
 
+#[cw_serde]
+pub struct TokenInfoProviderBase {
+    pub token_group_id: String,
+    pub denom: String,
+    pub ratio: Decimal,
+}
+
+impl TokenInfoProviderBase {
+    // Returns OK if the denom is same as the denom of the base Hydro token (e.g. ATOM, OSMO, etc.).
+    pub fn resolve_denom(&mut self, denom: String) -> StdResult<String> {
+        if self.denom != denom {
+            return Err(StdError::generic_err(format!(
+                "Mismatched denom: expected {}, got {}",
+                self.denom, denom
+            )));
+        }
+        Ok(self.token_group_id.clone())
+    }
+
+    pub fn get_token_group_ratio(&mut self) -> StdResult<Decimal> {
+        Ok(self.ratio)
+    }
+
+    pub fn get_all_token_group_ratios(&mut self) -> StdResult<HashMap<String, Decimal>> {
+        let mut ratios = HashMap::new();
+        ratios.insert(self.token_group_id.clone(), self.ratio);
+        Ok(ratios)
```

Comment:
Nitpick: Using `HashMap::from_iter([(self.token_group_id.clone(), self.ratio)])` would be simpler.